src/components/input/Input.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {
+  ForwardedRef,
+  forwardRef,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { Text, TextInput, TouchableOpacity, View } from 'react-native';
+import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
+
+import colors from '@constants/colors';
+
+import styles from './styles';
+
+// cprops is a shorthand for componentProps
+const Input = forwardRef(function Input(
```

Comment:
mention the use of `forwardRef`

Reply:
`forwardRef` is used here to forward the `ref` passed from the parent component to child component that is `TextInput`. We can't directly pass `ref` from user components to `react-native`'s native components.

Reply:
what is the use of this here with textInput?